src/researches/getPassiveVoice.js

```diff
@@ -1,110 +1,108 @@
- * @param {string} language The language to use for finding passive verbs.
- * @returns {boolean} True if passive is found, false if no passive is found.
+ * @param {Array} sentences Sentences extracted from the text.
+ * @param {string} language Language of the text.
+ * @returns {Object} The found passive sentences.
  */
-var determinePassives = function( sentencePart, language ) {
-	switch( language ) {
-		case "de":
-			sentencePart.setPassive( determinePassivesGerman( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries() ), "de" );
-			break;
-		case "fr":
-			sentencePart.setPassive( determinePassivesDefault( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), "fr" ) );
-			break;
-		case "es":
-			sentencePart.setPassive( determinePassivesDefault( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), "es" ) );
-			break;
-		case "en":
-		default:
-			sentencePart.setPassive( determinePassivesDefault( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), "en" ) );
-			break;
-	}
+const getMorphologicalPassives = function( sentences, language ) {
+	let passiveSentences = [];
+
+	forEach( sentences, function( sentence ) {
+		let strippedSentence = stripHTMLTags( sentence.getSentenceText() ).toLocaleLowerCase();
+
+		sentence.setPassive( isPassiveSentence( strippedSentence, language ) );
+
+		if ( sentence.isPassive() === true ) {
+			passiveSentences.push( sentence.getSentenceText() );
+		}
+	} );
+	return {
+		passiveSentences,
+	};
 };
 
 /**
- * Determines the number of passive sentences in the text.
+ * Looks for periphrastic passive voice.
  *
- * @param {Paper} paper The paper object to get the text from.
- * @returns {object} The number of passives found in the text and the passive sentences.
+ * @param {Array} sentences Sentences extracted from the text.
+ * @param {string} language Language of the text.
+ * @returns {Object} The found passive sentences.
  */
-module.exports = function( paper ) {
-	var text = paper.getText();
-	var locale = paper.getLocale();
-	var language = getLanguage( locale );
-	var sentences = getSentences( text );
-
-	var sentenceObjects = [];
+const getPeriphrasticPassives = function( sentences, language ) {
+	let passiveSentences = [];
 
 	forEach( sentences, function( sentence ) {
-		sentenceObjects.push( new Sentence( sentence, locale ) );
-	} );
-
-	var passiveSentences = [];
+		let strippedSentence = stripHTMLTags( sentence.getSentenceText() ).toLocaleLowerCase();
 
-	// Get sentence parts for each sentence.
-	forEach( sentenceObjects, function( sentence ) {
-		var strippedSentence = stripHTMLTags( sentence.getSentenceText() ).toLocaleLowerCase();
+		// The functionality based on sentencePart objects should be rewritten using array indices of stopwords and auxiliaries.
+		let sentenceParts = [];
 
-		var sentenceParts = getSentenceParts( strippedSentence, language );
+		if ( language === "de" ) {
```

Comment:
Where does this specific `de`/German check come from?
It's not clear from the code why this would be an exception, wondering if it should..

Reply:
Contrary to the other languages for which we have a passive voice assessment, German has a different definition of sentenceParts (YoastSEO.js/src/researches/german/passiveVoice/getSentenceParts.js). The whole sentenceParts business will be a subject of future improvements, so for now I decided to keep German as an exception.